browser-tests/utils/sharedTests.ts

```diff
@@ -0,0 +1,175 @@
+import shared from '../selectors/shared';
+
+export const yourSelection = async (
+  t: TestController,
+  testData: { choice2: string; choice1: string }
+) => {
+  const { heading, getHarborHeading, getUpButtonForHeading, nextButton } = shared.yourSelection;
+
+  await t.expect(heading.exists).ok();
+
+  await t.click(getUpButtonForHeading(getHarborHeading(`2. ${testData.choice2}`))).wait(500);
+
+  await t
+    .expect(getHarborHeading(`1. ${testData.choice2}`).exists)
+    .ok()
+    .expect(getHarborHeading(`2. ${testData.choice1}`).exists)
+    .ok();
+
+  await t.click(nextButton);
+};
+
+export const wsBoatInformation = async (
+  t: TestController,
+  testData: {
+    boatLength: string;
+    boatModel: string;
+    boatName: string;
+    boatRegistrationNumber: string;
+    boatType: string;
+    boatTypeIndex: string | undefined;
+    boatWidth: string;
+    trailerRegistrationNumber: string;
+  }
+) => {
+  const {
+    boatLength,
+    boatModel,
+    boatName,
+    boatRegistrationNumber,
+    boatStoredOnTrailer,
+    boatTypeSelect,
+    boatWidth,
+    heading,
+    nextButton,
+    trailerRegistrationNumber,
+  } = shared.boatInformation;
+
+  await t.expect(heading.exists).ok();
+
+  await t
+    .click(boatStoredOnTrailer)
+    .typeText(trailerRegistrationNumber, testData.trailerRegistrationNumber)
+    .typeText(boatRegistrationNumber, testData.boatRegistrationNumber)
+    .click(boatTypeSelect)
+    .click(boatTypeSelect.find('option').withText(testData.boatType))
+    .expect(boatTypeSelect.value)
+    .eql(testData.boatTypeIndex)
+    .click(boatWidth)
+    .pressKey('ctrl+a delete') // Clear width coming from first page of winter storage application
+    .typeText(boatWidth, testData.boatWidth)
+    .click(boatLength)
+    .pressKey('ctrl+a delete') // ditto for length
+    .typeText(boatLength, testData.boatLength)
+    .typeText(boatName, testData.boatName)
+    .typeText(boatModel, testData.boatModel);
+
+  await t.click(nextButton);
+};
+
+export const applicantInformation = async (
+  t: TestController,
+  testData: {
```

Comment:
should we create a proper types for test data(s)?

Reply:
Maybe 🤔

Reply:
Extracted the types